modules/codegen/src/main/scala/com/twilio/guardrail/generators/AkkaHttpServerGenerator.scala

```diff
@@ -213,28 +213,36 @@ object AkkaHttpServerGenerator {
     }
 
     def directivesFromParams(
-        required: Term => Type => Target[Term],
-        multi: Term => Type => Target[Term],
-        multiOpt: Term => Type => Target[Term],
-        optional: Term => Type => Target[Term]
+        required: Term => Type => Option[Term] => Target[Term],
+        multi: Term => Type => Option[Term] => Target[Term],
+        multiOpt: Term => Type => Option[Term] => Target[Term],
+        optional: Term => Type => Option[Term] => Target[Term]
```

Comment:
Any readability concerns with a function-function-function?

Reply:
I tried to maintain the level of precedence for relevancy. I have no objections to this style.

That being said, an alternative would be a tuple or a named parameter with explicit `unapply` overrides to prevent every field needing to be explicitly updated any time anything changes with the source API.